.github/workflows/nightly-builds.yml

```diff
@@ -0,0 +1,139 @@
+          server_port: 465
+          username: ${{secrets.MAIL_USERNAME}}
+          password: ${{secrets.MAIL_PASSWORD}}
+          subject: "Failed: ${{ github.workflow }} / ${{ github.job }}"
+          to: akka.official@gmail.com
+          from: Akka CI (GHActions)
+          body: |
+            Job ${{ github.job }} in workflow ${{ github.workflow }} of ${{github.repository}} failed!
+            https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+
+  jdk-nightly-tests:
+    name: JDK ${{ matrix.jdkVersion }} / Scala ${{ matrix.scalaVersion }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        # No need to specify the full Scala version. Only the Scala binary version
+        # is required and Akka build will set the right full version from it.
+        scalaVersion: ["2.12", "2.13"]
+        jdkVersion: ["adopt@1.8.0", "adopt@1.11"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK ${{ matrix.jdkVersion }}
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.jdkVersion }}
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.2
+
+      # Only run multijvm tests in Scala 2.13. No need to run it to
+      # multiple Scala versions AND multiple JDK versions. This will
+      # run multijvm tests to run for both JDK 8 and 11 though.
+      - name: Compile and Test
+        run: |-
+          ## TODO: restore the multi-node tests
+          sbt -jvm-opts .jvmopts-ci \
+            -Dakka.build.scalaVersion=${{ matrix.scalaVersion }} \
+            -Dakka.test.timefactor=2 \
+            -Dakka.cluster.assert=on \
+            -Dakka.test.multi-in-test=${{ matrix.scalaVersion == '2.13' }} \
+            -Dakka.test.tags.exclude=gh-exclude \
+            clean update Test/compile test checkTestsHaveRun
+
+      - name: Test Reports
+        # Makes it easier to spot failures instead of looking at the logs.
+        if: ${{ failure() }}
+        uses: marcospereira/action-surefire-report@v1
+        with:
+          report_paths: '**/target/test-reports/TEST-*.xml'
+          fail_if_no_tests: false
+          skip_publishing: true
+
+      - name: Docs
+        # Docs generation requires JDK 11. Checks with `startsWith` helps
+        # the check to be more resilient if the JDK version changes to a
+        # more specific one such as adopt@1.11.0-9.
+        if: ${{ startsWith(matrix.jdkVersion, 'adopt@1.11') }}
+        run: |-
+          sudo apt-get install graphviz
+          sbt -jvm-opts .jvmopts-ci \
+            -Dakka.build.scalaVersion=${{ matrix.scalaVersion }} \
+            -Dakka.genjavadoc.enabled=true \
+            doc
+
+      - name: Publish
+        run: |-
+          sbt -jvm-opts .jvmopts-ci \
+            -Dakka.build.scalaVersion=${{ matrix.scalaVersion }} \
+            publishLocal publishM2
```

Comment:
This in a matrix - does it mean we could publish a snapshot for Scala 2.13 while 2.12 failed? (Not entirely sure if that is a problem, just something I thought about)

Reply:
This is publishing local, so we won't have a problem with inconsistent snapshots unless I'm missing something from what you said.